### PR TITLE
feat: add notification banner to license view

### PR DESCRIPTION
**Pull request by jonathangoulding**
Add a notification banner, if the license will be in the next supplementary bull run. 

PRESROC notification 

Screenshot 2024-04-25 at 15 16 49

SROC notification
Screenshot 2024-04-25 at 15 21 11

Both charge scheme notifications
Screenshot 2024-04-25 at 15 21 56

### CODE REVIEW

**Comment by jonathangoulding** [comment]
Location of banner is assumed so will need to be reviewed by the reviewing developer.

File: app/views/licences/view.njk [hunk 1]
```diff
@@ -39,6 +40,19 @@
...
+    <div class="govuk-notification-banner" role="region" aria-labelledby="govuk-notification-banner-title" data-module="govuk-notification-banner">
+      <div class="govuk-notification-banner__header">
+        <h2 class="govuk-notification-banner__title" id="govuk-notification-banner-title">
+          Important
+        </h2>
+      </div>
+      <div class="govuk-notification-banner__content">
+     <strong> {{ notification }} </strong>
+      </div>
+    </div>
```
Comment (Cruikshanks) [inline app/views/licences/view.njk:53]: Unless we _really_ cannot make the Nunjucks version work, we always favour it. Our reasoning is when updates are made to the GOVUK package there is less need for us to revisit our views if we used the Nunjucks version.

So, could we try switching this to

```suggestion
    {% if notification %}
      {{ govukNotificationBanner({
        html: '<strong>' + notification + '</strong>'
      }) }}
    {% endif %}
```

> Testing locally this gave the same result.

File: test/presenters/licences/view-licence.presenter.test.js [hunk 2]
```diff
@@ -1073,6 +1074,52 @@ describe('View Licence presenter', () => {
       })
     })
   })
+  describe("the 'notification' property", () => {
+    describe('when the licence will not be in the next supplementary bill', () => {
```
Comment (Cruikshanks) [inline test/presenters/licences/view-licence.presenter.test.js:1078]: NIT

```suggestion
    describe('when the licence will not be in the next supplementary bill run', () => {
```

File: test/presenters/licences/view-licence.presenter.test.js [hunk 3]
```diff
@@ -1073,6 +1074,52 @@ describe('View Licence presenter', () => {
...
+      it('returns NULL', () => {
+        const result = ViewLicencePresenter.go(licence, licenceAbstractionConditions)
+
+        expect(result.notification).to.be.null()
+      })
+    })
+
+    describe('when the licence will be in the next supplementary bill (PRESROC)', () => {
```
Comment (Cruikshanks) [inline test/presenters/licences/view-licence.presenter.test.js:1086]:
```suggestion
    describe('when the licence will be in the next supplementary bill run (PRESROC)', () => {
```

File: test/presenters/licences/view-licence.presenter.test.js [hunk 4]
```diff
@@ -1073,6 +1074,52 @@ describe('View Licence presenter', () => {
...
+      })
+
+      it('returns the notification for PRESROC', () => {
+        const result = ViewLicencePresenter.go(licence, licenceAbstractionConditions)
+
+        expect(result.notification).to.equal('This license has been marked for the next supplementary bill run for the old charge scheme.')
+      })
+    })
+
+    describe('when the licence will be in the next supplementary bill (SROC)', () => {
```
Comment (Cruikshanks) [inline test/presenters/licences/view-licence.presenter.test.js:1098]:
```suggestion
    describe('when the licence will be in the next supplementary bill run (SROC)', () => {
```

File: test/presenters/licences/view-licence.presenter.test.js [hunk 5]
```diff
@@ -1073,6 +1074,52 @@ describe('View Licence presenter', () => {
...
+      })
+
+      it('returns the notification for SROC', () => {
+        const result = ViewLicencePresenter.go(licence, licenceAbstractionConditions)
+
+        expect(result.notification).to.equal('This license has been marked for the next supplementary bill run.')
+      })
+    })
+
+    describe('when the licence will be in the next supplementary bill (SROC & PRESROC)', () => {
```
Comment (Cruikshanks) [inline test/presenters/licences/view-licence.presenter.test.js:1110]:
```suggestion
    describe('when the licence will be in the next supplementary bill run (SROC & PRESROC)', () => {
```

**Review by Cruikshanks — CHANGES_REQUESTED** [review]
All of these requested changes are just a result of not knowing 'our ways' i.e. how @Cruikshanks likes things when he is paying attention 😬 . So, this is a ❤️ start and PR on day 2; 💪🏅!

One final thing to mention. There is no issue with using [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) when working on the branch.

But the first commit needs to follow [our conventions](https://github.com/DEFRA/software-development-standards/blob/master/processes/pull_requests.md#start-with-a-commit) and I encourage folks to use all 50 chars to make it as readable and meaningful as possible.

So, for this I'd be looking for something more like

```markdown
Add supp. billing notification to view licence

https://eaflood.atlassian.net/browse/WATER-4323

The legacy view licence page we are replacing displays a notification banner if the licence is flagged for supplementary billing. It could be flagged for PRESROC, SROC or both so the message has to reflect each possible state.

So, this means we need to do the same thing in our version of the view licence page.
```

If you follow our guide, GitHub does some cool stuff

- Your first `--allow-empty` commit will become your PR title and description
- When it comes to the squash & merge, assuming you haven't strayed _off-piste_ that initial commit message will be there to use again

Anyway, any questions or issues let me know.

File: app/views/licences/view.njk (hunk 1 above)
Comment (jonathangoulding) [reply app/views/licences/view.njk:53]: Mush better. Thanks.

File: .gitignore [hunk 6]
```diff
@@ -46,3 +46,6 @@ app/public/build
 
 # ignore husky config
 .husky/
+
+.idea/
+.webstorm/
```
[the remark this answers is not in the document]
Comment (jonathangoulding) [reply .gitignore]: i have removed for now. 

i will work on it locally for the mean time and move to a different change list until the tasks are ready to commit.

File: app/presenters/licences/view-licence.presenter.js [hunk 7]
```diff
@@ -29,7 +29,9 @@ function go (licence, licenceAbstractionConditions) {
     permitLicence,
     region,
     registeredTo,
-    startDate
+    startDate,
+    includeInPresrocBilling,
+    includeInSrocBilling
```
[the remark this answers is not in the document]
Comment (jonathangoulding) [reply app/presenters/licences/view-licence.presenter.js]: i have changed them all to alpha.

File: app/presenters/licences/view-licence.presenter.js [hunk 8]
```diff
@@ -68,7 +70,8 @@ function go (licence, licenceAbstractionConditions) {
     registeredTo,
     sourceOfSupply: abstractionDetails.sourceOfSupply,
     startDate: formatLongDate(startDate),
-    warning: _generateWarningMessage(ends)
+    warning: _generateWarningMessage(ends),
+    notification: _calculateNotificationBanner(includeInPresrocBilling, includeInSrocBilling)
```
[the remark this answers is not in the document]
Comment (jonathangoulding) [reply app/presenters/licences/view-licence.presenter.js]: i have changed them all to alpha.

File: app/views/licences/view.njk [hunk 9]
```diff
@@ -2,6 +2,7 @@
 {% from "govuk/components/back-link/macro.njk" import govukBackLink %}
 {% from "govuk/components/tabs/macro.njk" import govukTabs %}
 {% from "govuk/components/warning-text/macro.njk" import govukWarningText %}
+{% from "govuk/components/notification-banner/macro.njk" import govukNotificationBanner %}
```
[the remark this answers is not in the document]
Comment (jonathangoulding) [reply app/views/licences/view.njk]: i have changed them all to alpha.

**Review by Cruikshanks — APPROVED** [review]

**Pull request merged**